core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -224,6 +225,23 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
   def listBillingAccounts(): Future[PerRequestMessage] =
     gcsDAO.listBillingAccounts(userInfo) map(RequestComplete(_))
 
+  def getBillingProjectStatus(projectName: RawlsBillingProjectName): Future[PerRequestMessage] = {
+    val statusFuture: Future[Seq[RawlsBillingProjectStatus]] = for {
+      policies <- samDAO.getPoliciesForType(SamResourceTypeNames.billingProject, userInfo)
+      projectDetail <- dataSource.inTransaction { dataAccess => dataAccess.rawlsBillingProjectQuery.load(projectName) }
+    } yield {
+      policies.filter { policy =>
```

Comment:
My interpretation of this is, "check that this BP has at least one matching policy in Sam".

If so, you could choose to substitute`find` for `filter` and work with an`Option[RawlsBillingProjectStatus]` instead of a `Seq[RawlsBillingProjectStatus]`

Reply:
Yes, that could be cleaner, thanks!